streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -257,24 +275,39 @@
                     REPLICATION_FACTOR_DOC)
             .define(TIMESTAMP_EXTRACTOR_CLASS_CONFIG,
                     Type.CLASS,
-                    FailOnInvalidTimestamp.class.getName(),
+                    null,
```

Comment:
For backward compatibility, we need to keep the old default value.
Btw: we don't do any ordering here yet -- just above. so no need to reorder anything here.